synapse/storage/databases/state/bg_updates.py

```diff
@@ -101,9 +101,11 @@ def _get_state_groups_from_groups_txn(
             where_clause = " AND (%s)" % (where_clause,)
 
         if isinstance(self.database_engine, PostgresEngine):
+            # Suspicion start
             # Temporarily disable sequential scans in this transaction. This is
             # a temporary hack until we can add the right indices in
             txn.execute("SET LOCAL enable_seqscan=off")
+            # Suspicion end
```

Comment:
This thing that is part of `_get_state_groups_from_groups_txn` (the slowest part of `filter_events_for_client`) seems suspicious

This was introduced via https://github.com/matrix-org/synapse/pull/6296 (hasn't changed since it was introduced.